### DISCUSSION TITLE
How to run requests' tests?

### DISCUSSION THREAD

**Discussion by yjaaidi** [postmanlabs/postman-flows#2] [Q&A] [answered]
Hi,

First of all! Thank you for this amazing feature! It's quite exciting.

I was surprised that the selected requests' tests don't seem to be verified. 
Am I missing something?

Thanks!

**Comment by saswatds** [comment, marked as answer]
In Flows the failing of test's _do not block the execution of the flow_, Test is another _output_ of a request just like _response_.

If you wish to collect the test results in a summary then use the `Test Summary` block and you will be able to see the results in `Test Summary Panel`

![image](https://user-images.githubusercontent.com/8796447/136574260-95029d7d-fccc-4a00-bb8a-1040bfb94ef6.png)

Alternatively if you wish to take decision based on the test result, you could do something like this. 

![image](https://user-images.githubusercontent.com/8796447/136574185-38729a62-4d32-47af-a42d-55e55028aeb7.png)

**Comment by yjaaidi** [comment]
Cool! I didn't the see the test summary block.
It's helpful but I still would love some "test blocking" mode where failing tests would block the flow.

**Reply by yjaaidi** [reply]
Great job by the way!

**Reply by saswatds** [reply]
We can definitely consider a test blocking mode for the `Send Request` 💯. Looking for some more feedback from the community. 🤞🏽

**Reply by david-williams-mpp** [reply]
I just wrote a lengthy post while you were answering this so that kind of covers one of my points. Is this feature still in dev because I have the latest version and don't see the tests port or test summary block?

Can we maybe get a roadmap (even a simple list of bullets) so we have an idea what is in the pipeline to avoid raising an issue that is already solved?

Also +1 for having a pass/fail port

**Reply by saswatds** [reply]
The postman desktop app takes a long time to release hence these changes might show up later. But new changes are available in postman web everyday. You can access it here [Postman Web](http://go.postman.co/). 

Yeah, we are in the process of publishing both a Release Notes section and Roadmap here on Discussions. The adoptions of Flows has been beyond our expectation and we are a small team right now, so all the required docs should up within a week or so as we reprioritise and ramp up the team. Thanks a lot again for all the valuable feedback.

**Reply by MGMsystems** [reply]
Would be nice if you could have a test option on the validate block. 
If the validate block is false and the test option is on it could throw an error on the flow. Or mark as red. 
Would improve the use case for us on E2E testing with multiple cases and keeping the request clean from writing in the tests block.
![image](https://user-images.githubusercontent.com/4438834/137299163-e2e2e6ac-31a8-43ed-a054-1e3e3656856e.png)

**Reply by PericlesFeltrin** [reply]
@saswatds Do you have updates?

**Comment by Anusha-pru** [comment]
Hi,

I'm using postman flow, I have 5 tests associated to a request in my collection, i have added this request in the postman and pointed the tests to testsummary, trying to see the output of tests in the test summary but out of 5 only 2 tests are seen , is there a limit on how many tests can be run/seen in the test summary in postman flow?
Note: 
If have the tests like 1,2,3,4,5 in request, testsummary in flow is showing 1,2
If I change the order of tests to 3,4,5,2,1 in request, testsummary in flow is showing 3,4
so wondering is there a limit to show only top 2 ? also I can see all the tests running in the actual request and I made sure the request is saved. Can anyone help with this?

**Reply by saswatds** [reply]
Hey @Anusha-pru I don't remember putting any limits on the number of tests. I not been able to reproduce the issue as well. Are you using the web app or the desktop app? I pretty sure the tests are running, just not showing up in the UI.

**Reply by Anusha-pru** [reply]
Hi @saswatds , I'm using the desktop version and I came to know that this is a known issue in the postman desktop version, Is there a fix coming for this soon?

https://github.com/postmanlabs/postman-flows/discussions/248

**Reply by CharlieSnowCode** [reply]
I am noticing a similar issue.  We have 2 tests at the Collection level and 1 at the Request.  When I run the Flow, the Terminal and Test Summary only show the Collection level test results.  If I comment out the Collection level tests and run the Flow, I can see the test results from the Request.
